Single product template: changes to fit the new design

MaggieCabrera opened the pull request:
<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

Closes WOOTHME-76

This PR changes the single product template to fit better the new design. I didn't tweak it all the way because we need to merge the theme.json changes and because I'm unsure if this is the finalized version of this template. 

I have added the new blockified versions of the woo blocks and adapted the patterns to fit the design.

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

![Screenshot 2025-06-13 at 10-51-51 V-Neck T-Shirt – Musopress](https://github.com/user-attachments/assets/799f8eed-13b7-4bf2-8372-243f1f9273f2)

MaggieCabrera committed:
changed single template and patterns to fit the new design

MaggieCabrera commented:
Things to report on the Woo blocks:

For the Product Gallery:
- Thumbnails should have padding controls to be able to control separation between images
- Large image should also have some way to control the separation with the thumbnails
- I can change the aspect ratio of the thumbnail images but I can't change the ratio on the large image, this is inconsistent
- I can't insert any other blocks, why can't I insert group/columns block or spacers here?      

Add to cart + options:

- If the template is forcing certain settings they can no longer be overriden by theme.json, they need to be changed by a template. This is the case for font size of `add-to-cart-with-options-variation-selector-attribute-name`. It would be better if we set the default value via block.json instead.
- Nitpick: the html validator through an error on the quantity selector button because there is no space before type: `<button aria-label="Reduce quantity of V-Neck T-Shirt"type="button" data-wp-on--click="actions.decreaseQuantity" class="wc-block-components-quantity-selector__button wc-block-components-quantity-selector__button--minus">` There are a few other errors in this area too:

<img width="2781" alt="Screenshot 2025-06-12 at 13 09 49" src="https://github.com/user-attachments/assets/70fd3d67-5a28-48da-baab-8850478e420f" />


Other blocks: 
- Price block should allow for different colors/typography controls when the price is on sale, to give different weight to the old vs new prices, this is a really common pattern
- Product summary block: changing the color via theme.json works on the frontend but not on the site editor
- Accordion panel: changing padding settings on the editor works fine, but changing them in theme.json makes the panel visible even if it's closed for some reason!
- No styling possible for the reviews form, and the spacing between elements is very tight.

/cc @kmanijak @Aljullu I will be updating this comment as I work on the template and open up the necessary issues once I'm done, but I wanted to give you a heads up just in case
/cc @mikachan in case you see low hanging fruit for you (I think there are a few)

Aljullu commented:
Thanks for highlighting this. Looking forward to knowing all the limitations you face and which things can be improved from the WooCommerce blocks side.

> - If the template is forcing certain settings they can no longer be overriden by theme.json, they need to be changed by a template. This is the case for font size of `add-to-cart-with-options-variation-selector-attribute-name`. It would be better if we set the default value via block.json instead.

Heads-up that we updated the font sizes of the _Add to Cart + Options_ template parts, and [we just removed the hard-coded font size](https://github.com/woocommerce/woocommerce/pull/58653/files#diff-b56e5361dec7b611eb028a7ac9e024633e7632a1dfddc284012f3622b052c5f4L16) on the `add-to-cart-with-options-variation-selector-attribute-name` block. I think that's what you were referring to?

kmanijak commented:
Thanks for feedback @MaggieCabrera!

> For the Product Gallery:
> Thumbnails should have padding controls to be able to control separation between images
> Large image should also have some way to control the separation with the thumbnails

These are good ideas. Unfortunately I'm AFK June 11 - 20 so won't be able to work on it. Tagging @sunyatasattva and @ellenbauer to prioritise these.

> I can change the aspect ratio of the thumbnail images but I can't change the ratio on the large image, this is inconsistent

This is something we wanted and there's even [an issue for it](https://linear.app/a8c/issue/WOOPLUG-4113/product-gallery-allow-changing-an-aspect-ratio-of-large-image) but it wasn't considered high priority and needed for first iteration of starter theme. Also, we didn't want to work on it as we're utilising Product Image inside Large Image and that change the implementation. Once done we can try working on it but not sure if that can be achieved by WooCommerce 10.0 feature freeze.

> I can't insert any other blocks, why can't I insert group/columns block or spacers here?

Where exactly? Between Thumbnails and Large Image? I guess the original design was to keep it consistent and the block usage is limited via `allowedBlocks`. Unfortunately blocks are coupled here and until we reimplement the block completely we cannot allow for full flexibility.

kmanijak commented:
> For the Product Gallery:
> - Thumbnails should have padding controls to be able to control separation between images
> - Large image should also have some way to control the separation with the thumbnails

> These are good ideas. Unfortunately I'm AFK June 11 - 20 so won't be able to work on it. Tagging @sunyatasattva and @ellenbauer to prioritise these.

Re this ☝️, @shivanarrthine, would you have some suggestions how these controls could look like? My understanding is we should not (or even _couldn't_) use native `supports.padding` option since that padding would target different blocks. Maybe we could consider using `blockGap` but need to verify if that would be applicable 🤔

MaggieCabrera commented:
> Heads-up that we updated the font sizes of the _Add to Cart + Options_ template parts, and [we just removed the hard-coded font size](https://github.com/woocommerce/woocommerce/pull/58653/files#diff-b56e5361dec7b611eb028a7ac9e024633e7632a1dfddc284012f3622b052c5f4L16) on the `add-to-cart-with-options-variation-selector-attribute-name` block. I think that's what you were referring to?

That is pretty recent, thanks for pointing that out!

MaggieCabrera commented:
> Where exactly? Between Thumbnails and Large Image? I guess the original design was to keep it consistent and the block usage is limited via `allowedBlocks`. Unfortunately blocks are coupled here and until we reimplement the block completely we cannot allow for full flexibility.

Yes, my intention here was to make more space between them and I couldn't in any way, I couldn't via spacing settings and I couldn't add a spacer or put the two blocks inside a column or group block

Thanks for the quick feedback @Aljullu and @kmanijak, I don't think any of this is a blocker for v1 of the theme, If I ever found something that would be, I'd open a linear issue right away and mark it as high priority and flag for urgency, but I wanted to let you all know for things like what Albert mentioned, that would have escaped me if he didn't mention that was already taken care of.

MaggieCabrera commented:
> Re this ☝️, @shivanarrthine, would you have some suggestions how these controls could look like? My understanding is we should not (or even _couldn't_) use native `supports.padding` option since that padding would target different blocks. Maybe we could consider using `blockGap` but need to verify if that would be applicable 🤔

Block gap would definitely make sense for me here

MaggieCabrera committed:
renamed pattern, made some changes to fit new design

MaggieCabrera committed:
revamped product area to fit the new design

MaggieCabrera committed:
moved reviews out of the accordion

MaggieCabrera committed:
created pattern for related products

MaggieCabrera committed:
made returns info into a template part

MaggieCabrera committed:
fitted patterns to design

MaggieCabrera committed:
renamed containers and reformatted for readibility

MaggieCabrera committed:
breadcrumbs and product reviews markup

MaggieCabrera committed:
better spacing for reviews

MaggieCabrera committed:
changes to the reviews title

MaggieCabrera commented:
I've updated this PR with the latest changes in the design. They were extensive. Not everything that was designed was possible as outlined [here](https://github.com/woocommerce/woo-themes/pull/97#issuecomment-2956154869) but I think we got quite close! This is ready for a review

mikachan commented on woostarter/patterns/about-left-aligned-image-right-text-dark.php:
```diff
@@ -17,7 +17,7 @@
 <div class="wp-block-columns alignwide are-vertically-aligned-center"><!-- wp:column {"verticalAlignment":"center"} -->
 <div class="wp-block-column is-vertically-aligned-center"><!-- wp:image {"id":61,"aspectRatio":"1","scale":"cover","sizeSlug":"full","linkDestination":"none"} -->
-<figure class="wp-block-image size-full"><img src="<?php echo esc_url( get_template_directory_uri() ); ?>/assets/images/wp-content/themes/woostarter/assets/images/coming-soon-img.webp" alt="A person holding a rustic, handmade ceramic mug in a cozy, dimly lit setting." class="wp-image-61" style="aspect-ratio:1;object-fit:cover"/></figure>
+<figure class="wp-block-image size-full"><img src="<?php echo esc_url( get_template_directory_uri() ); ?>/assets/images/coming-soon-img.webp" alt="A person holding a rustic, handmade ceramic mug in a cozy, dimly lit setting." class="wp-image-61" style="aspect-ratio:1;object-fit:cover"/></figure>
```
Should the alt text be translatable here? I've noticed there a few other cases where similar alt text is not wrapped in a translation function too.

Are we going to address things like this as part of https://github.com/woocommerce/woo-themes/issues/67 later on?

MaggieCabrera replied:
if the pattern is only half translated, I would fix it in the PR, #67 is for the ones that are fully untranslated

MaggieCabrera replied:
Fixed for the patterns on this PR

mikachan commented on woostarter/theme.json:
```diff
@@ -339,15 +339,44 @@
+						"style": "solid",
+						"width": "1px"
+					},
+					"color": null,
+					"style": null,
+					"width": null
```
Are these set to `null` to override the accordion item's default styles?

MaggieCabrera replied:
I think CBT did this for me, I'll check if it's needed or not

MaggieCabrera replied:
Seems like they could be deleted, done! Thanks for the review

mikachan reviewed (approved):
This is looking great to me across lots of different resolutions. Nice work!

Mobile:

https://github.com/user-attachments/assets/310365d1-467c-4f25-861b-882cf9e54d6a


Desktop:

https://github.com/user-attachments/assets/237d9a4f-fb95-4e3c-8303-81d47cd86773

I've left a couple of inline comments but they're more questions and definitely not blockers.

MaggieCabrera committed:
moved main tag out of the pattern, fixed translations for alt texts

MaggieCabrera committed:
fix attr function on alt tags

MaggieCabrera committed:
removed extra style rules from accordion on theme.json
